### PR TITLE
sft updates

### PR DESCRIPTION
**Pull request by holygits**
TODO:
- [x] fix merge conflict
- [x] resolve listings check

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ async function main() {
                 zip(
                     api.query.genericAsset.stakingAssetId(),
                     api.query.genericAsset.spendingAssetId(),
-                ).subscribe(([stakingAssetId, spendingAssetId, devices]) => {
+                ).subscribe(([stakingAssetId, spendingAssetId]) => {
```
Comment (holygits) [inline test/index.js:24]: unused var

File: packages/types/src/interfaces/nft/token-id.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import { Struct } from "@polkadot/types";
```
Comment (holygits) [inline packages/types/src/interfaces/nft/token-id.ts:1]: not used yet, might be useful to return this type from derived queries

**Review by KarishmaBothara — APPROVED** [review]
lgtm

File: packages/api/test/e2e/nft.e2e.ts [hunk 3]
```diff
@@ -49,59 +51,150 @@ afterAll(async () => {
...
+            tokenId = data[1];
+            console.log(`got token: ${tokenId}`);
+          }
         });
-        // first token has id 0
-        let tokenAttributes = (await api.query.nft.tokenAttributes(collectionId, 0));
-        expect(tokenAttributes.toJSON()).toEqual(attributes);
-        expect((await api.query.nft.tokenOwner(collectionId, 0)).toString()).toBe(tokenOwner.address);
+        let seriesId = tokenId[1];
+        let seriesAttributes = (await api.query.nft.seriesAttributes(collectionId, seriesId));
```
Comment (KarishmaBothara) [inline packages/api/test/e2e/nft.e2e.ts:92]: this failed on my local with 
Error: nft.seriesAttributes(CollectionId, SeriesId) is a doublemap, requiring 2 arguments, 1 found

File: packages/types/src/interfaces/nft/enhanced-token-id.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    seriesId: string | number,
+    serialNumber: string | number,
+}
+
+export class EnhancedTokenId extends Struct implements Codec {
+    constructor(registry: Registry, value?: TokenIdValue | [number, number, number] | any) {
+      super(
+        registry,
+        {
+            collectionId: 'u32',
```
Comment (KarishmaBothara) [inline packages/types/src/interfaces/nft/enhanced-token-id.ts:17]: may be we just have them of the type
collectionId: CollectionId
seriesId: SeriesId
serialNumber: SerialNumber from ./types

**Pull request merged**